core/templates/services/html-length.service.ts

```diff
@@ -71,70 +78,67 @@ export class HtmlLengthService {
     const sanitizedHtml = this.sanitizer.sanitize(
       SecurityContext.HTML, htmlString) as string;
     let totalCharacters =
-    this.calculateBaselineLength(sanitizedHtml, 'character');
+    this.calculateBaselineLength(sanitizedHtml, CALCULATION_TYPE_CHARACTER);
     // Identify custom tags using regex on the original HTML string.
-    const customTagRegex = /<oppia-noninteractive-(?:math|image)[^>]*>/g;
-    const customTags = htmlString.match(customTagRegex);
+    const customTags = htmlString.match(CUSTOM_TAG_REGEX);
 
     if (customTags) {
       for (const customTag of customTags) {
         totalCharacters += this.getWeightForNonTextNodes(
-          customTag, 'character');
+          customTag, CALCULATION_TYPE_CHARACTER);
       }
     }
     return totalCharacters;
   }
 
   private calculateBaselineLength(
-      sanitizedHtml: string, wordOrCharacter: string): number {
+      sanitizedHtml: string, calculationType: CalculationType): number {
     let domparser = new DOMParser();
     let dom = domparser.parseFromString(sanitizedHtml, 'text/html');
     let totalWeight = 0;
     for (let tag of Array.from(dom.body.children)) {
       const ltag = tag.tagName.toLowerCase();
       if (this.textTags.includes(ltag)) {
         totalWeight +=
-        this.getWeightForTextNodes(tag as HTMLElement, wordOrCharacter);
+        this.getWeightForTextNodes(tag as HTMLElement, calculationType);
       }
     }
     return totalWeight;
   }
 
   private getWeightForTextNodes(
-      textNode: HTMLElement, wordOrCharacter: string): number {
+      textNode: HTMLElement, calculationType: CalculationType): number {
     const textContent = textNode.textContent || '';
 
-    if (wordOrCharacter === 'word') {
-      const words = textContent.trim().split(' ');
-      const wordCount = words.length;
-      return wordCount;
+    let textContentTrimmed = textContent.trim();
+    let textContentCount: number = 0;
+    if (calculationType === CALCULATION_TYPE_WORD) {
+      const words = textContentTrimmed.split(' ');
+      textContentCount = words.length;
+    } else if (calculationType === CALCULATION_TYPE_CHARACTER) {
+      const characters = textContentTrimmed.split('\n').join('').split('');
+      textContentCount = characters.length;
     }
-    const characters = textContent.trim().split('\n').join('').split('');
-    let characterCount = characters.length;
-    return characterCount;
+    return textContentCount;
   }
 
   private getWeightForNonTextNodes(
-      nonTextNode: string, wordOrCharacter: string): number {
+      nonTextNode: string, calculationType: CalculationType): number {
     if (nonTextNode.includes('oppia-noninteractive-math')) {
       return 1;
     }
     const altTextMatch = nonTextNode
```

Comment:
Add newline above this one.

Can you add a test for what happens if the html contains both a math RTE component and an image component? Also one that contains 2 of one and 1 of the other?

Reply:
@seanlip On adding a blank new line above line no. 126 gives a linting error that blank lines can't be used for padding.

I have added a test case when both the image and maths components are present and a separate case of either was already present: https://github.com/oppia/oppia/pull/19518/files#diff-faa264bf77420ce5737ca9c8d025f37f4a23848af3c92b17aa08111b9255fc7eR214

Reply:
I'm asking for a newline above `const altTextMatch`, not the part above it.

Reply:
Done!!